src/components/Budget/BudgetTarget/BudgetTarget.tsx

```diff
@@ -77,49 +76,70 @@ function BudgetTargetProjectDialog(props: IBudgetTargetProjectProps) {
 		UPDATE_PROJECT_BUDGET_TARGET
 	);
 
-	const { data: orgCurrencies } = useQuery(GET_ORG_CURRENCIES_BY_ORG, {
-		variables: {
-			filter: {
-				organization: dashboardData?.organization?.id,
-				isHomeCurrency: true,
-			},
-		},
+	let { fetchData: getOrgCurrencies, data: orgCurrencies } = useLazyQueryCustom({
+		query: GET_ORG_CURRENCIES_BY_ORG,
 	});
 
-	const { data: budgetCategory } = useQuery(GET_ORGANIZATION_BUDGET_CATEGORY, {
-		variables: {
-			filter: {
-				organization: dashboardData?.organization?.id,
-			},
-		},
+	let { fetchData: getBudgetCategory, data: budgetCategory } = useLazyQueryCustom({
+		query: GET_ORGANIZATION_BUDGET_CATEGORY,
 	});
 
-	const { data: donors } = useQuery(GET_PROJ_DONORS, {
-		variables: {
-			filter: {
-				project: dashboardData?.project?.id,
-			},
-		},
+	let { fetchData: getDonors, data: donors } = useLazyQueryCustom({
+		query: GET_PROJ_DONORS,
 	});
 
+	useEffect(() => {
+		if (dashboardData?.organization) {
+			getOrgCurrencies({
+				filter: {
+					organization: dashboardData?.organization?.id,
+					isHomeCurrency: true,
+				},
+			});
+		}
+	}, [getOrgCurrencies, dashboardData]);
+
+	useEffect(() => {
+		if (dashboardData?.organization) {
+			getBudgetCategory({
+				filter: {
+					organization: dashboardData?.organization?.id,
+				},
+			});
+		}
+	}, [getBudgetCategory, dashboardData]);
+
+	useEffect(() => {
+		if (dashboardData?.project) {
+			getDonors({
+				filter: {
+					project: dashboardData?.project?.id,
+				},
+			});
+		}
+	}, [getDonors, dashboardData]);
+
```

Comment:
We can add dashboardData?.project in the dependency array . It will run only when this particular value is changed

Reply:
React will give warning that it has missing dependency dashboardData

Reply:
if you write `[dashboardData?.project]` ? It shouldn't

Reply:
I did that earlier, it was giving the warning

Reply:
okay let it be then